Sources/Core/EthereumABI/ABIElements.swift

```diff
@@ -325,7 +325,7 @@ extension ABI.Element.Function {
             // set a flag to detect the request succeeded
         }
 
-        if returnArray.isEmpty {
+        if returnArray.isEmpty && !outputs.isEmpty {
```

Comment:
@JeneaVranceanu @yaroslavyaroslav can you please check this line as I'm not too familiar with this part of the code and also don't know much about how tested it is.

Reply:
a SOL function such as:
function fund(uint256 _value) public {
        addressToAmountFunded[msg.sender] += _value;
        funders.push(msg.sender);
    }

will have empty return array and it's outputs are empty so in this situation we should return success not nil

Reply:
Ok cool and now make a test for this. ;D

Because I think this is way too important parts of the project to just change a condition without even looking/knowing the different cases. I don't know if we already have tests for this, maybe @JeneaVranceanu knows as he liked the reply.